### PR TITLE
USAGOV-1721-mobile-menu-translation-fix: Set variables['node'] in preprocess_block

### PR DESCRIPTION
**Pull request by mdranove**
## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-1721

## Description

Updated usagov_preprocess_block in usagov.theme to pass $variables['node'] to twig templates.

## Type of Changes

- [ ] New Feature
- [ X ] Bugfix
- [ ] Frontend (Twig, Sass, JS)
  - Add screenshot showing what it should look like
- [ ] Drupal Config (requires "drush cim")
- [ ] New Modules (requires rebuild)
- [ ] Documentation
- [ ] Infrastructure
  - [ ] CMS
  - [ ] WAF
  - [ ] Egress
  - [ ] Tools
- [ ] Other

## Testing Instructions

1. In English, open the mobile navigation menu on the homepage and a handful of other pages.
2. The navigation menu appears in English and the links are aliased (not like 'node/33')
3. Repeat for Spanish.

### Change Requirements

- [ ] Requires New Documentation (Link: {})
- [ ] Requires New Config
- [ ] Requires New Content

## Security Review

- [ ] Adds/updates software (including a library or Drupal module)
- [ ] Communication with external service
- [ ] Changes permissions or workflow
- [ ] Requires SSPP updates

## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
4. Find the commit of this pull request.
5. Build and deploy the changes.
6. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.

### CODE REVIEW

File: web/themes/custom/usagov/usagov.theme [hunk 1]
```diff
@@ -295,9 +295,15 @@ function usagov_preprocess_block(&$variables) {
...
-  $entity = \Drupal::routeMatch()->getParameter('node') ?? \Drupal::routeMatch()
-    ->getParameter('taxonomy_term');
-  $variables['entity'] = $entity;
+  if (\Drupal::routeMatch()->getParameter('node')) {
+    $entity = \Drupal::routeMatch()->getParameter('node');
+    $variables['node'] = $entity;
+  }
+  elseif (\Drupal::routeMatch()
+    ->getParameter('taxonomy_term')) {
+    $entity = \Drupal::routeMatch()
```
Comment (omerida) [inline web/themes/custom/usagov/usagov.theme:304]: Do you need a `$variables['entity'] = $entity` to pass the term to twig templates?

Comment (mdranove) [reply web/themes/custom/usagov/usagov.theme:304]: Good question but for now not needed. Might come up later though as the new wizard mobile nav still needs to be built.

Comment (mdranove) [reply web/themes/custom/usagov/usagov.theme:304]: Added a return statement as I was seeing php errors on user pages.

**Review by akf — APPROVED** [review]
Tested locally and, as expected, this fixes the problem. Thanks!

**Pull request merged**